[MNT] Setup uv (#529)
neurobagel/api: README.md

### Python
#### Install dependencies

After cloning the repository, install the package in editable mode with development dependencies using uv:

```bash
uv sync --all-groups
```

This will create a virtual environment (if one doesn't exist), install the package in editable mode, and install all dependency groups including dev dependencies.

#### Launch the API

> To launch the API make sure you're in the repository's main directory and that your `.env` file is configured.

Reviewer: ```suggestion
To launch the API, make sure you're in the repository root and that your [`.env` file has been configured](#set-the-environment-variables).
```

    file.write(create_query(return_agg=False))
```

> ## We manage dependencies with `uv`

Reviewer: ```suggestion
#### Updating dependencies
```
I'd then move this entire section under `### Python`, at the end of the section
